Mission involves creep.lua

```diff
@@ -0,0 +1,86 @@
+local argsUtil = require("Module:ArgsUtil")
+local cargoUtil = require("Module:CargoUtil")
+
+local p = {}
+
+-- kill ranges based on difficulty level
+local killRanges = {
+    easy = { 4, 12 },
+    hard = { 2, 5 },
+    special = { 4, 8 }
+}
+
+-- Function to calculate the reward range based on the kill range
+local function calculateRewardRange(lowerBound, upperBound)
```

Comment:
often devs like to use a function before it's defined. this is nice because it means you can read top-to-bottom, like "ah I see that I will be using calculateRewardRange, now what is it that calculateRewardRange does?"

lua does not exactly let you use functions before they're used if you scope them locally like this, but what you CAN DO is say

```lua
-- use myFunction
p.main()
		p.myFunction()
end

function p.myFunction() end
```

Because the functions are all scoped as members of the table `p` it magically works.

In reality, I like to do something like this:

```lua
local p = {}
local h = {} -- h stands for 'helpers'

function p.main()
    h.myHelperFunction()
end


function h.myHelperFunction() end
```